src/server/service/kv.rs

```diff
@@ -625,20 +625,16 @@ impl<T: RaftStoreRouter<RocksEngine> + 'static, E: Engine, L: LockManager> Tikv
                 RAFT_MESSAGE_RECV_COUNTER.inc();
                 let to_store_id = msg.get_to_peer().get_store_id();
                 if to_store_id != store_id {
-                    future::err(Error::from(RaftStoreError::StoreNotMatch {
```

Comment:
I think this is intended. Otherwise no way to recover from a wrong resolved address.